server/neptune/admin/server.go

```diff
@@ -0,0 +1,269 @@
+		clientCreator,
+		config.GithubCfg.TemporalAppInstallationID,
+		config.DataDir,
+		featureAllocator,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "initializing github activities")
+	}
+
+	cronScheduler := internalSync.NewCronScheduler(config.CtxLogger)
+
+	server := Server{
+		Logger:        config.CtxLogger,
+		CronScheduler: cronScheduler,
+		Crons: []*internalSync.Cron{
+			{
+				Executor:  crons.NewRuntimeStats(scope).Run,
+				Frequency: 1 * time.Minute,
+			},
+		},
+		HTTPServerProxy:     httpServerProxy,
+		Port:                config.ServerCfg.Port,
+		StatsScope:          scope,
+		StatsCloser:         statsCloser,
+		TemporalClient:      temporalClient,
+		TerraformActivities: terraformActivities,
+		TerraformTaskQueue:  config.TemporalCfg.TerraformTaskQueue,
+		GithubActivities:    githubActivities,
+	}
+	return &server, nil
+}
+
+func (s Server) Start() error {
+	defer s.shutdown()
+
+	ctx := context.Background()
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		terraformWorker := s.buildTerraformWorker()
+		if err := terraformWorker.Run(worker.InterruptCh()); err != nil {
+			log.Fatalln("unable to start terraform worker", err)
+		}
+
+		s.Logger.InfoContext(ctx, "Shutting down terraform worker, resource clean up may still be occurring in the background")
+	}()
+
+	// Ensure server gracefully drains connections when stopped.
+	stop := make(chan os.Signal, 1)
+	// Stop on SIGINTs and SIGTERMs.
+	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
+
+	s.Logger.Info(fmt.Sprintf("Atlantis started - listening on port %v", s.Port))
+
+	go func() {
+		err := s.HTTPServerProxy.ListenAndServe()
+
+		if err != nil && err != http.ErrServerClosed {
+			s.Logger.Error(err.Error())
+		}
+	}()
+
+	<-stop
+	wg.Wait()
+
+	return nil
+}
+
+func (s Server) shutdown() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := s.HTTPServerProxy.Shutdown(ctx); err != nil {
+		s.Logger.Error(err.Error())
+	}
+
+	s.TemporalClient.Close()
+
+	// flush stats before shutdown
+	if err := s.StatsCloser.Close(); err != nil {
+		s.Logger.Error(err.Error())
+	}
+
+	s.Logger.Close()
+}
+
+// Note that we will need to do things similar to how gateway does it to get the metadata we need
+// specifically the root
+
+func (s Server) buildTerraformWorker() worker.Worker {
+	// pass the underlying client otherwise this will panic()
+	terraformWorker := worker.New(s.TemporalClient.Client, s.TerraformTaskQueue, worker.Options{
+		Interceptors: []interceptor.WorkerInterceptor{
+			temporal.NewWorkerInterceptor(),
+		},
+		// default is 1k, however Nish's comment says that:
+		// "30 falls in line with our current cpu resourcing for each worker and should be the
+		// maximum number of concurrent tf operations we allow."
+		// We shouldn't need too many as we are only doing one thing here on this batch pod
+		MaxConcurrentActivityExecutionSize: 30,
```

Comment:
nit: we can remove this comment too, but sounds good. I think it's okay if we start off with this value and better tune it later on. I don't think we are a risk of overuse here.